packages/skill-lesson/path-to-purchase/pricing.tsx

```diff
@@ -109,11 +109,14 @@ export const Pricing: React.FC<React.PropsWithChildren<PricingProps>> = ({
       purchaseId: formattedPrice?.upgradeFromPurchaseId,
     })
 
-  const isRestrictedUpgrade = purchaseToUpgrade?.status === 'Restricted'
-
   const defaultCoupon = formattedPrice?.defaultCoupon
   const appliedMerchantCoupon = formattedPrice?.appliedMerchantCoupon
 
+  const isRestrictedUpgrade =
+    purchaseToUpgrade?.status === 'Restricted' &&
+    appliedMerchantCoupon &&
+    appliedMerchantCoupon.type !== 'ppp'
```

Comment:
this is for the "remove restricted access" scenario, which isn't in this PR yet because still noodling...